src/content/developers/tutorials/yellow-paper-evm/index.md

```diff
@@ -0,0 +1,396 @@
+| ----: | --------- | - | - | ----------- |
+| 0x31  | BALANCE   | 1 | 1 | Get balance of the given account.
+||||| ...
+
+The address whose balance we need to find is *μ<sub>s</sub>[0] mod 2<sup>160</sup>*.
+The top of the stack is the address, but because addresses are only 160 bits, we calculate the value [modulo](https://en.wikipedia.org/wiki/Modulo_operation) 2<sup>160</sup>.
+
+If *σ[μ<sub>s</sub>[0] mod 2<sup>160</sup>] ≠ ∅*, it means that there is information about this address.
+In that case, *σ[μ<sub>s</sub>[0] mod 2<sup>160</sup>]<sub>b</sub>* is the balance for that address.
+If *σ[μ<sub>s</sub>[0] mod 2<sup>160</sup>] = ∅*, it means that this address is uninitialized and the balance is zero.
+You can see the list of account information fields in section 4.1 on p. 4. 
+
+The second equation, *A'<sub>a</sub> ≡ A<sub>a</sub> ∪ {μ<sub>s</sub>[0] mod 2<sup>160</sup>}*, is related to the difference in cost between access to warm storage (storage that has recently been accessed and is likely to be cached) and cold storage (storage that hasn't been accessed and is likely to be in slower storage that is more expensive to retrieve). 
+*A<sub>a</sub>* is the list of addresses previously accessed by the transaction, which should therefore be cheaper to access, as defined in section 6.1 on p. 8.
+You can read more about this subject in [EIP-2929](https://eips.ethereum.org/EIPS/eip-2929).
+
+
+| Value | Mnemonic  | δ  | α  | Description |
+| ----: | --------- | -- | -- | ----------- |
+| 0x8F  | DUP16     | 16 | 17 | Duplicate 16th stack item.
+||||| *μ′<sub>s</sub>[0] ≡ μ<sub>s</sub>[15]*
+
+Note that to use any stack item, we need to pop it, which means we also need to pop all the stack items on top of it.
+In the case of [`DUP<n>`](https://www.evm.codes/#8f) and [`SWAP<n>`](https://www.evm.codes/#9f), this means having to pop and then push up to sixteen values.
+
+
+## 9.5 (The Execution Cycle)
+
+Now that we have all the parts, we can finally undertand how the execution cycle of the EVM is documented.
+
+Equation (155) says that given the state:
+
+- *σ* (global blockchain state) 
+- *μ* (EVM state)
+- *A* (substate, changes to happen when the transaction ends)
+- *I* (execution environment)
+
+The new state is *(σ', μ', A', I')*.
+
+Equations (156)-(158) define the stack and the change in it due to an opcode (*μ<sub>s</sub>*). 
+Equation (159) is the change in gas (*μ<sub>g</sub>*). 
+Equation (160) is the change in the program counter (*μ<sub>pc</sub>*).
+Finally, equations (161)-(164) specify that the other parameters stay the same, unless explicitly changed by the opcode.
+
+With this the EVM is fully defined.
+
+
+## Conclusion
+
+Mathematical notation is precise, and has allowed the Yellow Paper to specify every detail of Ethereum.
+However, it does have some drawbacks:
+
+- It can only be understood by humans, which means that [compliance tests](https://github.com/ethereum/tests) have to be written manually.
```

Comment:
```suggestion
- It can only be understood by humans, which means that [compliance tests](https://github.com/ethereum/tests) must be written manually.
```